sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { generate_uuid } from "rhea-promise";
+import { isTokenCredential, TokenCredential } from "@azure/core-auth";
+import {
+  ServiceBusClientOptions,
+  createConnectionContextForTokenCredential,
+  createConnectionContextForConnectionString
+} from "./constructorHelpers";
+import { ConnectionContext } from "./connectionContext";
+import { ClientEntityContext } from "./clientEntityContext";
+import { ClientType } from "./client";
+import { Sender } from "./sender";
+import { GetSessionReceiverOptions, ContextWithSettlement } from "./models";
+import { Receiver, ReceiverImpl, SubscriptionRuleManagement } from "./receivers/receiver";
+import { SessionReceiver, SessionReceiverImpl } from "./receivers/sessionReceiver";
+
+export class ServiceBusClient {
+  private _connectionContext: ConnectionContext;
+
+  /**
+   *
+   * @param connectionString A connection string for Azure Service Bus.
+   * NOTE: this connection string can contain an EntityPath, which is ignored.
```

Comment:
We don't validate the entity name provided in sender/receiver then? 
Maybe better to add validation at the sender/receiver level?

Reply:
Filed #7816